### PR TITLE
feat(clarity): implement Clarity types and serialization

### LINKED ISSUE #2: Implement Clarity Value Representation
Clarity is the smart contract language used by the Stacks blockchain. Unfortunately, Golang is unsupported by Stacks so we need to create an equivalent to the [Stacks JS transactions package](https://github.com/hirosystems/stacks.js/tree/main/packages/transactions). This enables constructing contract calls, parsing transaction results, and handling cross-chain messages.

To interact with Stacks smart contracts and process transaction data, we need to accurately represent Clarity values in our codebase. This involves creating a type-safe representation of Clarity values in Golang, along with serialization and deserialization methods in accordance with the [Clarity value representation specification](https://github.com/stacksgov/sips/blob/main/sips/sip-005/sip-005-blocks-and-transactions.md#clarity-value-representation).

**Clarity types**
Create a `ClarityValue` interface that all Clarity types will implement.

Implement the following types, each implementing the ClarityValue interface:

-  `IntValue` for Clarity int
-  `UIntValue` for Clarity uint
-  `BoolValue` for Clarity bool
-  `BufferValue` for Clarity buffer
-  `StringASCIIValue` for Clarity string-ascii
-  `StringUTF8Value` for Clarity string-utf8
-  `PrincipalValue` for Clarity principal (both standard and contract principals)
-  `ResponseValue` for Clarity response
-  `OptionalValue` for Clarity optional
-  `ListValue` for Clarity list
-  `TupleValue` for Clarity tuple

**Serialization**
Add a `Serialize() ([]byte, error)` method to the ClarityValue interface.

Implement the `Serialize()` method for each Clarity type. This is the 1-byte type ID followed by the serialized payloads defined in the [Clarity value representation specification](https://github.com/stacksgov/sips/blob/main/sips/sip-005/sip-005-blocks-and-transactions.md#clarity-value-representation).

**Deserialization**
Add a `Deserialize([]byte, error)` method to the ClarityValue interface. This method verifies the type ID byte and deserializes the remaining bytes into the ClarityValue according to the type-specific format defined in the [Clarity value representation specification](https://github.com/stacksgov/sips/blob/main/sips/sip-005/sip-005-blocks-and-transactions.md#clarity-value-representation).

**Constructors**
Create constructors for each Clarity type (`NewIntValue(int)`, `NewListValue([]ClarityValue)`, etc.) and a factory function `NewClarityValue([]byte) (ClarityValue, error)` that takes in the serialized Clarity value bytes, reads the type ID byte, constructs the appropriate Clarity value type, and calls the type's Deserialize method on the serialized payload.

**Acceptance Criteria**

- All Clarity types can be represented in Golang with type safety
- Serialization produces byte sequences that comply with the Stacks specification
- The `NewClarityValue` factory function correctly instantiates and deserializes any valid Clarity value
- Unit tests cover all types and edge cases with >80% code coverage

### PR DESCRIPTION
**Pull request by CyrusVorwald**
Closes #2

### CODE REVIEW

File: pkg/clarity/types.go [hunk 1]
```diff
@@ -0,0 +1,658 @@
...
+func (o *OptionSome) Type() ClarityType {
+	return ClarityTypeOptionSome
+}
+
+func (o *OptionSome) Serialize() ([]byte, error) {
+	valueBytes, err := o.Value.Serialize()
+	if err != nil {
+		return nil, err
+	}
+	return append([]byte{byte(ClarityTypeOptionSome)}, valueBytes...), nil
```
Comment (AntonAndell) [inline pkg/clarity/types.go:314]: In other files you followed space after bracket formatting. Might be worth to be consistent.

**Review by AntonAndell — APPROVED** [review]
Can be merged unless we want to change formatting slightly. Looks good though. Hard to catch detailed errors though. But structure looks good

File: pkg/clarity/types.go (hunk 1 above)
Comment (CyrusVorwald) [reply pkg/clarity/types.go:314]: I'm not sure I follow.

image

Comment (AntonAndell) [reply pkg/clarity/types.go:314]: Aah just formatting that after a bracket there is a new line
like 

}

"next here"
